src/components/Heading/Heading.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+import { InjectedProps, withTheme } from '../../hocs/withTheme'
+
+export interface HeadingProps {
+  children: string
+  type?: 'ScreenTitle' | 'SectionTitle' | 'BlockTitle' | 'SubBlockTitle' | 'SubSubBlockTitle'
+  tag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+}
+
+const HeadingComponent: React.FC<HeadingProps & InjectedProps> = ({ theme, ...props }) => (
+  <Wrapper
+    as={props.tag ? props.tag : 'h1'}
+    theme={theme}
+    type={props.type ? props.type : 'ScreenTitle'}
+  >
+    {props.children}
+  </Wrapper>
+)
```

Comment:
The prospect is bad because there are many conditional branches in the component.
How about assigning a default value to an argument?

```tsx
const HeadingComponent: React.FC<HeadingProps & InjectedProps> = ({ tag = 'h1', type = 'ScreenTitle', children, theme }) => (
  <Wrapper as={tag} type={type} theme={theme}>
    {children}
  </Wrapper>
)
```

Reply:
thx 😍